lib/landrush/cap/host/windows/configure_visibility_on_host.rb

```diff
@@ -19,46 +20,82 @@ def configure_visibility_on_host(env, ip, tlds)
-          #    IP Address:                           10.10.10.1
-          #    Subnet Prefix:                        10.10.10.0/24 (mask 255.255.255.0)
-          #    InterfaceMetric:                      10
-          #    DNS servers configured through DHCP:  None
-          #    Register with which suffix:           Primary only
-          #    WINS servers configured through DHCP: None
-          # \n\n
-          # ...
-          def get_network_name(ip)
-            cmd_out = `netsh interface ip show config`
-            network_details = cmd_out.split(/^$/).reject(&:empty?).select do |settings|
-              begin
-                lines = settings.split(/\n/).reject(&:empty?)
-                subnet = lines[3]
-                next false unless subnet =~ /Subnet Prefix/
-
-                mask = IPAddr.new(subnet.match(%r{.* (\d{1,3}\.\d{1,3}\.\d{1,3}.\d{1,3}/\d{1,3}).*}).captures[0])
-                address = IPAddr.new(ip)
-
-                mask.include?(address)
-              rescue StandardError
-                false
+          # Given an IP determines the network adapter guid, if any.
+          # We can't use netsh due to its output being locale dependant.
+          # Using Windows registry we achieve the same result of netsh in
+          # whatever language Windows is.
+          def get_network_guid(address)
+            address = IPAddr.new(address.to_s)
+
+            interfaces do |interface_guid|
+              interface = open_interface("#{INTERFACES}\\#{interface_guid}")
+              if_ip, if_mask = interface_address(interface)
+
+              next if if_ip.nil?
+
+              if_net = IPAddr.new("#{if_ip}/#{if_mask}")
+              return interface_guid if if_net.include?(address)
+            end
+          rescue StandardError
+            nil
+          end
+
+          private
+
+          # Fetch interfaces from registry
+          # This is a separate method to make testing easier
+          def interfaces
+            Win32::Registry::HKEY_LOCAL_MACHINE.open(INTERFACES) do |interfaces|
+              interfaces.each_key do |guid, _|
+                yield guid
               end
             end
-            return nil if network_details[0].nil? || network_details[0].empty?
+          end
 
-            network_details[0].split(/\n/).reject(&:empty?)[0].match(/Configuration for interface "(.*)"/).captures[0].strip
+          # Opens an interface key from registry
+          # This is a separate method to make testing easier
+          def open_interface(guid_path)
+            Win32::Registry::HKEY_LOCAL_MACHINE.open(guid_path)
           end
 
-          private
+          # Fetches IP/mask info from registry from a given registry entry
+          # This is a separate method to make testing easier
+          def interface_address(interface)
+            dhcp_enabled = interface.read('EnableDHCP')[1]
+
+            if dhcp_enabled
+              if_ip = interface.read('DhcpIPAddress')[1]
```

Comment:
Interesting. Did you by any chance test this against a DHCP enabled Windows box as well as one with a static IP?

Reply:
Yes, I did. :) That's, actually, how I came across this. :)

Reply:
Cool